src/luascript.cpp

```diff
@@ -7077,6 +7078,26 @@ int LuaScriptInterface::luaCreatureGetHealth(lua_State* L)
 	return 1;
 }
 
+int LuaScriptInterface::luaCreatureSetHealth(lua_State* L)
+{
+	// creature:setHealth(health)
+	Creature* creature = getUserdata<Creature>(L, 1);
+	if (!creature) {
+		lua_pushnil(L);
+		return 1;
+	}
+
+	creature->health = std::max<int32_t>(0, getNumber<uint32_t>(L, 2));
```

Comment:
Shouldn’t it update `healthMax` as well in case `health` exceeds `healthMax`?

Reply:
Actually it should be `creature->health = std::min<int32_t>(creature->health, getNumber<uint32_t>(L, 2));`